catroid/src/main/java/org/catrobat/catroid/visualplacement/VisualPlacementTouchListener.java

```diff
@@ -23,50 +23,88 @@
 
 package org.catrobat.catroid.visualplacement;
 
-import android.os.SystemClock;
 import android.view.MotionEvent;
 import android.widget.ImageView;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class VisualPlacementTouchListener {
 
 	private Mode mode;
-	private float previousY;
+	private static final long DELAY_THRESHOLD = 30;
+	private static final float DISTANCE_THRESHOLD = 10.f * 10.f;
 	private float previousX;
-	private long lastTimeMove = 0;
+	private float previousY;
+	private long touchDownTimestamp = 0;
+	private TimeProvider timeProvider;
+	private List<TouchEventData> eventData = new ArrayList<>();
+
+	private final class TouchEventData {
+		private final long timeStamp;
+		private final float xCoordinate;
+		private final float yCoordinate;
+
+		private TouchEventData(long time, float x, float y) {
+			timeStamp = time;
+			xCoordinate = x;
+			yCoordinate = y;
+		}
+	}
 
 	private void setMode(Mode mode) {
 		this.mode = mode;
 	}
 
+	public VisualPlacementTouchListener(TimeProvider timeProvider) {
+		this.timeProvider = timeProvider;
+	}
+
 	public boolean onTouch(ImageView imageView, MotionEvent event, CoordinateInterface coordinateInterface) {
 		if (event.getPointerId(0) == 0) {
 			float currentX = event.getRawX();
 			float currentY = event.getRawY();
+			long currentTimestamp = timeProvider.elapsedRealtime();
```

Comment:
Oh neat, with this we don't need the interface to avoid powermocking.
I changed it in my repo but getDownTime() only gives us the timestamp of the touchdown event, but getEventTime() fits here pretty well.